tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc

```diff
@@ -6928,66 +6928,21 @@ NodeDef MakePadNodeDef(std::string name, DataType dtype) {
   return pad.operation.node()->def();
 }
 
-template <typename CType>
 struct PadTestParams {
   std::vector<int> input_dims;
   std::vector<int> pad_dims;
-  std::vector<CType> input_values;
+  std::vector<float> input_values;
   std::vector<int> expected_output_dims;
-  std::vector<CType> expected_output_values;
+  std::vector<float> expected_output_values;
+  Status status;
 };
 
-template <DataType dtype>
-void TestConvertPad(OpConverterTest* test) {
-  typedef typename EnumToDataType<dtype>::Type CType;
-
-  std::vector<PadTestParams<CType>> params{
-      {
-          /*input_dims=*/{1, 2, 1},  // H, W, C
-          /*pad_dims=*/{4, 2},       // #dims, {pad_before, pad_after}
-          /*input_values=*/CastTestVector<float, CType>({2.0f, -1.0f}),
-          /*expected_output_dims=*/{2, 3, 1},  // H, W, C
-          /*expected_output_values=*/
-          CastTestVector<float, CType>({0.0, 0.0, 0.0, 2.0f, -1.0f, 0.0}),
-      },
-  };
-
-  for (int i = 0; i < params.size(); ++i) {
-    test->Reset();
-    // Create pad node.
-    NodeDef node_def = MakePadNodeDef("my_pad", dtype);
-    // Create input tensor
-    nvinfer1::DataType trt_type;
-    TF_ASSERT_OK(TfTypeToTrtType(dtype, &trt_type));
-    test->AddTestTensor("input", params[i].input_dims, /*batch_size=*/1,
-                        /*trt_dtype=*/trt_type);
-    // Create output size.
-    test->AddTestWeights<int32>("padding", params[i].pad_dims,
-                                {0, 0, 1, 0, 0, 1, 0, 0});
-    test->RunValidationAndConversion(node_def);
-
-    TRT_TensorOrWeights output;
-    TF_EXPECT_OK(test->GetTensorOrWeights("padding", &output));
-
-    // Create input data for tensors.
-    const DataVec input_data{
-        {"input", test->AsTensor<CType>(params[i].input_values)}};
-    DataVec output_data{
-        {"my_pad", test->ConstructTensor<CType>(
-                       params[i].expected_output_values.size())}};
-
-    TF_EXPECT_OK(test->BuildAndRun(input_data, &output_data));
-    ExpectArrayAlmostEqual(params[i].expected_output_values,
-                           GetSpanForData<CType>(output_data[0]), CType(1e-5));
-  }
-}
-
-TEST_F(OpConverterTest, ConvertPad) {
+TEST_P(OpConverterTest2, ConvertPad) {
```

Comment:
You will need to update this line before merging. Or it will fail: `OpConverter_FP32_FP16_Test`